### PR TITLE
IN-6 Updated swagger to match latest ADR structural agreeements

### PR DESCRIPTION
**Pull request by jackgoodby**
# Description

Updated swagger file with the various change implications from recent ADRs

# Issues Resolved

[List any existing issues this PR resolves]

# Contribution Checklist

- [ ] Terraform plans
- [ ] New functionality has been documented in the README if applicable

### CODE REVIEW

File: docs/architecture/swagger.v2.yaml [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+          example: '2019-01-01'
+        reporting_period_to:
+          type: string
+          format: date
+          example: '2019-12-31'
+        year:
+          type: integer
+          format: int32
+          minimum: 2010
+          maximum: 2020
```
Comment (gregtyler) [inline docs/architecture/swagger.v2.yaml]: Seems a bit risky to codify this

File: docs/architecture/swagger.v2.yaml [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+              example: '/data/attributes/payment_status'
+    ReportAttributes:
+      type: object
+      required:
+        - source
+        - reporting_period_from
+        - reporting_period_to
+        - year
+        - date_submitted
+      properties:
```
Comment (gregtyler) [inline docs/architecture/swagger.v2.yaml]: We should probably add a `type`property of "Property and Finance" or "Health and Welfare" (though I'd want to clarify those options with PMs/Sirius Devs). It's not very well supported in DigiDeps right now, but a client is likely to have one of each report each year.

File: docs/architecture/swagger.v2.yaml [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+          multipart/form-data:
+            schema:
+              type: object
+              properties:
+                report:
+                  type: object
+                  required:
+                    - data
+                    - links
+                  properties:
```
Comment (gregtyler) [inline docs/architecture/swagger.v2.yaml]: Is this right? I'd assume we just need to send the `ReportAttributes` object

File: docs/architecture/swagger.v2.yaml [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+              example: 'https://opg-data-api.service.gov.uk/help/errors/forbidden'
+        source:
+          type: object
+          properties:
+            parameter:
+              type: string
+              example: 'clientref'
+            pointer:
+              type: string
+              example: '/data/attributes/payment_status'
```
Comment (gregtyler) [inline docs/architecture/swagger.v2.yaml]: I'm not quite sure what this is referring to?

File: docs/architecture/swagger.v2.yaml [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+    ReportAttributes:
+      type: object
+      required:
+        - source
+        - reporting_period_from
+        - reporting_period_to
+        - year
+        - date_submitted
```
Comment (gregtyler) [inline docs/architecture/swagger.v2.yaml]: An erroneous report would be unsubmitted by the case manager, corrected and resubmitted by the deputy. In that case, we'd probably make the request again with the same source, report dates and year (though a different date submitted).

My understanding is that Sirius would find the same report and figure out whether to overwrite the original report, add a duplicate etc.

Is that correct? Or do we need to do something special for resubmissions?

File: docs/architecture/swagger.v2.yaml (hunk 2 above)
Comment (jackgoodby) [reply docs/architecture/swagger.v2.yaml]: Checked with Liz and she agrees on this. Added.

File: docs/architecture/swagger.v2.yaml [hunk 6]
```diff
@@ -0,0 +1,636 @@
...
+            $ref: '#/components/schemas/ReportId'
+      requestBody:
+        required: true
+        description: The supporting document
+        content:
+          image/*:
+            schema:
+              type: string
+              format: binary
+          application/*:
```
Comment (gregtyler) [inline docs/architecture/swagger.v2.yaml]: Should this just be PDF? I'm not aware of other application subtypes we'd accept
```suggestion
          application/pdf:
```

Comment (jackgoodby) [reply docs/architecture/swagger.v2.yaml]: Hey - this is for the supportingdocs endpoint, which I believe can be any sort of file that the user uploads?

Comment (gregtyler) [reply docs/architecture/swagger.v2.yaml]: Our restrictions are that it must be one of:
- application/pdf
- application/x-pdf
- image/png
- image/jpeg

I've no issue with the API being more liberal, but thought I should check.

File: docs/architecture/swagger.v2.yaml [hunk 7]
```diff
@@ -0,0 +1,636 @@
...
+                    $ref: '#/components/schemas/ReportSupportingDocument'
+                  links:
+                    type: object
+                    properties:
+                      api-docs:
+                        type: string
+                        format: uri
+                        example: 'https://opg-data-api.service.gov.uk/developer/'
+                  meta:
+                    type: object
```
Comment (gregtyler) [inline docs/architecture/swagger.v2.yaml]: This suggests that the only valid response will be a 200. I assume the error responses are the same as the other endpoint (with 404 accounting for a bad report ID?

File: docs/architecture/swagger.v2.yaml (hunk 6 above)
Comment (jackgoodby) [reply docs/architecture/swagger.v2.yaml]: That's good that it's so limited - I believe we'll be sending financial csvs through this endpoint as well now though?

Comment (gregtyler) [reply docs/architecture/swagger.v2.yaml]: Great point, that's `text/csv` also then

**Pull request merged**